sys/include/auto_init_macros.h

```diff
@@ -0,0 +1,425 @@
+#define CONCAT(a, b) CONCAT_HELP(a, b)
+/** @} */
+#endif
+
+/**
+ * @brief   Auto init function type
+ */
+typedef void(*auto_init_fn)(void);
+
+/**
+ * @name    Priority levels of auto-initialized modules from highest to lowest
+ *
+ * If you have a custom module that needs to be initialized after some other module,
+ * for example after `ztimer`, you can add another priority level like that:
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~{.c}
+ * #include "auto_init_macros.h"
+ *
+ * #define AUTO_INIT_PRIO_MY_MODULE AUTO_INIT_PRIO_AFTER(AUTO_INIT_PRIO_ZTIMER, 0)
+ * AUTO_INIT_MODULE(my_module, AUTO_INIT_PRIO_MY_MODULE, my_module_init);
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * @ref AUTO_INIT_MODULE
+ * @internal
+ *
+ * @experimental
+ * @{
+ */
+#define AUTO_INIT_PRIO_0                            a_
+#define AUTO_INIT_PRIO_1                            b_
+#define AUTO_INIT_PRIO_2                            c_
+#define AUTO_INIT_PRIO_3                            d_
+#define AUTO_INIT_PRIO_4                            e_
+#define AUTO_INIT_PRIO_5                            f_
+#define AUTO_INIT_PRIO_6                            g_
+#define AUTO_INIT_PRIO_7                            h_
+#define AUTO_INIT_PRIO_8                            i_
+#define AUTO_INIT_PRIO_9                            j_
+#define AUTO_INIT_PRIO_10                           k_
+#define AUTO_INIT_PRIO_11                           l_
+#define AUTO_INIT_PRIO_12                           m_
+#define AUTO_INIT_PRIO_13                           n_
+#define AUTO_INIT_PRIO_14                           o_
+#define AUTO_INIT_PRIO_15                           p_
+#define AUTO_INIT_PRIO_16                           q_
+#define AUTO_INIT_PRIO_17                           r_
+#define AUTO_INIT_PRIO_18                           s_
+#define AUTO_INIT_PRIO_19                           t_
+#define AUTO_INIT_PRIO_20                           u_
+#define AUTO_INIT_PRIO_21                           v_
+#define AUTO_INIT_PRIO_22                           w_
+#define AUTO_INIT_PRIO_23                           x_
+#define AUTO_INIT_PRIO_24                           y_
+#define AUTO_INIT_PRIO_25                           z_
+#define AUTO_INIT_PRIO_26                           za_
+#define AUTO_INIT_PRIO_27                           zb_
+#define AUTO_INIT_PRIO_28                           zc_
+#define AUTO_INIT_PRIO_29                           zd_
+#define AUTO_INIT_PRIO_30                           ze_
+#define AUTO_INIT_PRIO_31                           zf_
+#define AUTO_INIT_PRIO_32                           zg_
+#define AUTO_INIT_PRIO_33                           zh_
+#define AUTO_INIT_PRIO_34                           zi_
+#define AUTO_INIT_PRIO_35                           zj_
+#define AUTO_INIT_PRIO_36                           zk_
+#define AUTO_INIT_PRIO_37                           zl_
```

Comment:
Some Doxygen comments need silencing.

Reply:
Actually I don´t understand why `#defines` inside
```
/**
 * @name Group name
 * @{
 */

/** @} */
```
appear to be undocumented for Doxygen.
How do I silence Doxygen? Do I have to wrap it within
```
#if !defined(DOXYGEN)

#define STUFF

#endif
```
?

Or should I better add a pseudo documentation?
` #define STUFF /**< some stuff */`